frontend/src/bundles/edit-template/store/actions.ts

```diff
@@ -1,39 +1,35 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
-import { type TemplateGetAllItemResponseDto } from 'shared/build/index.js';
 
-import { type AsyncThunkConfig } from '~/bundles/common/types/types.js';
+import { type AsyncThunkConfig } from '~/bundles/common/types/async-thunk-config.type';
 
-import {
-    type TemplateUpdateItemRequestDto,
-    type TemplateUpdateItemResponseDto,
-} from '../types/types.js';
+import { type TemplateDto } from '../../templates-page/types/types';
 import { name as sliceName } from './slice.js';
 
-type EditTemplayePayload = Partial<TemplateUpdateItemRequestDto> & {
-    id: string;
-};
+const getTemplateById = createAsyncThunk<TemplateDto, string, AsyncThunkConfig>(
+    `${sliceName}/get-by-id`,
+    async (id, { extra }) => {
+        const { templateApi } = extra;
+        return await templateApi.getTemplateById(id);
```

Comment:
```suggestion
        return templateApi.getTemplateById(id);
```

Reply:
async func should have await statement

Reply:
Why it should be async than?

Reply:
because its a async fetch request

Reply:
Outside of try/catch blocks, return await is redundant.